internal/common/armadaerrors/errors.go

```diff
@@ -548,3 +590,32 @@ func NewCombinedErrPodUnschedulable(errs ...error) *ErrPodUnschedulable {
 	}
 	return result
 }
+
+// ErrUnauthorized represents an error that occurs when a client tries to
+// perform some action through the gRPC API for which it cannot authenticate.
+//
+// It may be necessary populate the Action field by recovering this error at
+// the gRPC endpoint (using errors.As) and updating the field in-place.
+type ErrUnauthorized struct {
```

Comment:
Have you tried this to check the error returned to the user? Please give some examples.

Reply:
`2022/10/04 14:10:52 error submitting request &api.JobSubmitRequest{Queue:"test", JobSetId:"job-set-1", JobRequestItems:[]*api.JobSubmitRequestItem{(*api.JobSubmitRequestItem)(0xc0005b00a0)}}: rpc error: code = Unauthenticated desc = [x-request-id: "JLdcWD3VE5CmFqUoCzpfz"] Could not authorize user "<none>" via service "Basic" while attempting action "/api.Submit/SubmitJobs"; rpc error: code = Unauthenticated desc = invalid username/password` Looks good from `armadactl`'s view.